teuthology/orchestra/remote.py

```diff
@@ -518,49 +520,59 @@ def move_file(self, src, dst, sudo=False, mode=None, owner=None,
             args += ' && ' + chown + ' ' + owner + ' ' + dst
         self.run(args=args)
 
+    @contextmanager
+    def _maybe_copy_from(self, sudo, path):
+        try:
+            if sudo:
+                temp_path = self.mktemp()
+                self.sh(f'sudo cp ${path} ${temp_path}')
+                self.sh(f'sudo chmod 0666 ${temp_path}')
+                yield temp_path
+            else:
+                yield path
+        finally:
+            if sudo:
+                self.sh(f'rm ${temp_path}')
+
     def read_file(self, path, sudo=False, stdout=None,
-                              offset=0, length=0):
+                              offset=0, length=-1):
```

Comment:
What was wrong with this?

Reply:
for 3 reasons.

1. i am trying to follow the convention of `read(size)` of a file-like object in python.
2. simplify the code
3. for better readability. as i don't understand the use case of `length=0`.

Reply:
In case of 1. we should point this in change description why it is changed, looks good to me to align with python convention, but it is not necessary.
For 2. I don't see simplification.
For 3. Why not, it is the same as length negative, this is exactly the same.